packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1829,36 +1860,36 @@ export class TransactionController extends BaseControllerV1<
     address: string,
     networkClientId?: NetworkClientId,
   ): Promise<NonceLock> {
-    let nonceMutexForChainId: Mutex | undefined;
+    let releaseLockForChainIdKey: (() => void) | undefined;
     let { nonceTracker } = this;
     if (networkClientId && this.enableMultichain) {
       const networkClient = this.getNetworkClientById(networkClientId);
-      nonceMutexForChainId = this.nonceMutexByChainId.get(
-        networkClient.configuration.chainId,
-      );
+      releaseLockForChainIdKey = await this.#acquireNonceLockForChainIdKey({
```

Comment:
total nit but maybe `releaseLockForChainIdAndAddress`? The key is less clear, why not just spell out what constitutes the lock "combination"? Same would go for `#acquireNonceLockForChainIdKey` -> `#acquireNonceLockForChainIdAndAddress`